usecases/sharding/remote_index.go

```diff
@@ -270,6 +276,27 @@ func (ri *RemoteIndex) SearchShard(ctx context.Context, shard string,
 	return r.first, r.second, node, err
 }
 
+func (ri *RemoteIndex) VectorDistanceForQuery(ctx context.Context, shard string,
+	id strfmt.UUID,
+	targetVectors []string,
+	searchVectors [][]float32,
+	replEnabled bool,
+) ([]float32, string, error) {
+	f := func(node, host string) (interface{}, error) {
+		scores, err := ri.client.VectorDistanceForQuery(ctx, host, ri.class, shard,
+			id, targetVectors, searchVectors)
+		if err != nil {
+			return nil, err
+		}
+		return scores, err
+	}
+	rr, node, err := ri.queryReplicas(ctx, shard, f)
+	if err != nil {
+		return nil, node, err
+	}
+	return rr.([]float32), node, err
```

Comment:
**suggestion:** `rr.([]float32)` this could panic wdyt if we do check if it's if this type 
```go
rrs, ok :=   rr.([]float32)
if !ok {
   return nil, node, fmt.Errorf("it's not of type ....")
}
return rrs, node, err
```

Reply:
added (I copied that whole part from another function - so it is probably missing there, too)